src/scripts/main.js

```diff
@@ -1,3 +1,396 @@
+
+option4.value = 'New York';
+option4.textContent = 'New York';
+
+option5.value = 'Edinburgh';
+option5.textContent = 'Edinburgh';
+
+option6.value = 'San Francisco';
+option6.textContent = 'San Francisco';
+
+selectOffice.append(option1);
+selectOffice.append(option2);
+selectOffice.append(option3);
+selectOffice.append(option4);
+selectOffice.append(option5);
+selectOffice.append(option6);
+
+const labelAge = document.createElement('label');
+const inputAge = document.createElement('input');
+
+labelAge.textContent = 'Age: ';
+inputAge.name = 'age';
+inputAge.type = 'number';
+inputAge.dataset.qa = 'age';
+form.append(labelAge);
+labelAge.append(inputAge);
+inputAge.required = true;
+
+const labelSalary = document.createElement('label');
+const inputSalary = document.createElement('input');
+
+labelSalary.textContent = 'Salary: ';
+inputSalary.name = 'salary';
+inputSalary.type = 'number';
+inputSalary.dataset.qa = 'salary';
+form.append(labelSalary);
+labelSalary.append(inputSalary);
+
+const button = document.createElement('button');
+
+button.type = 'submit';
+button.textContent = 'Save to table';
+
+form.append(button);
+document.body.append(form);
+
+const pushNotification = (posTop, posRight, title, description, type) => {
+  const message = document.createElement('div');
+  const header = document.createElement('h2');
+  const content = document.createElement('p');
+
+  message.dataset.qa = 'notification';
+  message.className = `notification ${type}`;
+  header.className = 'title';
+
+  header.append(title);
+  content.append(description);
+  message.append(header);
+  message.append(content);
+
+  message.style.top = `${posTop}px`;
+  message.style.right = `${posRight}px`;
+
+  document.body.append(message);
+
+  setTimeout(() => {
+    message.remove();
+  }, 2000);
+};
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const data = Object.fromEntries(new FormData(form));
+
+  if (data.salary < 1) {
+    pushNotification(200, 10, 'Invalid salary',
+      'Input correct salary.', 'error');
+
+    return;
+  }
+
+  const cellName = data.name;
+  const position = data.position;
+  const office = data.office;
+  const age = data.age;
+  const salary = `$${parseInt(data.salary).toLocaleString('en-US')}`;
+
+  if (cellName.length < 4) {
+    pushNotification(200, 10, 'Invalid name',
+      'The name must consist of at least 4 characters.', 'error');
+
+    return;
+  };
+
+  if (age < 18 || age > 90) {
+    pushNotification(200, 10, 'Invalid age',
+      'Age must be more than 18 and less than 90.', 'error');
+
+    return;
+  }
+
+  tbody.insertAdjacentHTML('beforeend', `
+    <tr>
+      <td>${cellName}</td>
+      <td>${position}</td>
+      <td>${office}</td>
+      <td>${age}</td>
+      <td>${salary}</td>
+     </tr>
+  `);
+
+  pushNotification(100, 10, 'The operation is successful',
+    'Employee was successfully added', 'success');
+});
+
+tbody.addEventListener('dblclick', e => {
```

Comment:
What about to move commands for every targetCell case into separated functions and call them?  It would be easier to read